aws/rust-runtime/aws-http/src/content_encoding.rs

```diff
@@ -276,6 +282,30 @@ where
     len
 }
 
+fn total_rendered_length_of_trailers(header_map: Option<&HeaderMap>) -> u64 {
```

Comment:
Seems like we should have a unit test that serializes the header map with the `http` crate and compares its length against this function. Especially with multiple headers of the same name.

Reply:
I added a test that ensures our trailer serializer and trailer length calculator stay in sync